lib/model/model.go

```diff
@@ -2195,6 +2196,20 @@ func (m *model) ConnectedTo(deviceID protocol.DeviceID) bool {
 	return ok
 }
 
+// Check if folderstate is valid for a particular device (i.e not paused)
+func (m *model) IsFolderPausedOnDevice(folder string, device protocol.DeviceID) bool {
+	m.mut.RLock()
+	defer m.mut.RUnlock()
+
+	if states, ok := m.remoteFolderStates[device]; ok {
+		if state, ok := states[folder]; ok {
+			return state != remoteFolderValid
```

Comment:
Hi Simon,

I vaguely recall this being added (or extended) in the last couple of years to show remoteFolderPaused so I assume that older versions of Syncthing will return valid regardless of paused state as long as the folder is shared?

folderstate.go
```
func (s remoteFolderState) String() string {
	switch s {
	case remoteFolderUnknown:
		return "unknown"
	case remoteFolderNotSharing:
		return "notSharing"
	case remoteFolderPaused:
		return "paused"
	case remoteFolderValid:
		return "valid"
	default:
		return "unknown"
	}
}

```

Reply:
Not entirely sure and no time right now to investigate. Generally there's no need to avoid a solution that doesn't work for very old clients. So as long as you default to "assume available" on unclear/missing states, that should be all good.

Reply:
Thanks, I'll check and make sure that's the case.

Reply:
`remoteFolderValid` is an enumeration constant which was introduced in #8201 and #8283, replacing the previous `Accepted` boolean flag on `type FolderCompletion` and the `model.remotePausedFolders` set (now `model.remoteFolderStates`).

I don't think there's any chance of incompatibility here, as that didn't change the protocol.  It just changed the way the detected remote folder state is stored in the model.